packages/knip/src/plugins/react-router/index.ts

```diff
@@ -0,0 +1,60 @@
+import { existsSync } from 'node:fs';
+import type { IsPluginEnabled, Plugin, ResolveEntryPaths } from '../../types/config.js';
+import { toEntry } from '../../util/input.js';
+import { join } from '../../util/path.js';
+import { hasDependency, load } from '../../util/plugin.js';
+import type { PluginConfig, RouteConfigEntry } from './types.js';
+
+// https://reactrouter.com/start/framework/routing
+
+const title = 'react-router';
+
+const enablers = ['@react-router/dev'];
+
+const isEnabled: IsPluginEnabled = ({ dependencies }) => hasDependency(dependencies, enablers);
+
+const config: string[] = ['react-router.config.{js,ts}'];
+
+const entry: string[] = [];
+
+const production: string[] = [];
+
+const resolveEntryPaths: ResolveEntryPaths<PluginConfig> = async (localConfig, options) => {
+  const { configFileDir } = options;
+  const appDirectory = localConfig.appDirectory ?? 'app';
+  const appDir = join(configFileDir, appDirectory);
+
+  let routeConfig: RouteConfigEntry[] = [];
+
+  const routesPathTs = join(appDir, 'routes.ts');
+  const routesPathJs = join(appDir, 'routes.js');
+
+  if (existsSync(routesPathTs)) {
+    routeConfig = await load(routesPathTs);
+  } else if (existsSync(routesPathJs)) {
+    routeConfig = await load(routesPathJs);
```

Comment:
maybe good idea to check if `routeConfig` is actually set later on?

Reply:
I initialize `routeConfig` as an empty array above. So if no routes.ts file is defined it is ignored.

Reply:
Ah, gotcha. Good!